src/js/components/help/downloads.js

```diff
@@ -0,0 +1,323 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { connect } from 'react-redux';
+import copy from 'copy-to-clipboard';
+import Cookies from 'universal-cookie';
+
+import { Accordion, AccordionDetails, AccordionSummary, Chip, Menu, MenuItem, Typography } from '@mui/material';
+import { ArrowDropDown, ExpandMore, FileDownloadOutlined as FileDownloadIcon, Launch } from '@mui/icons-material';
+
+import { setSnackbar } from '../../actions/appActions';
+import { getCurrentUser, getDocsVersion, getIsEnterprise, getTenantCapabilities } from '../../selectors';
+import Time from '../common/time';
+import { getToken } from '../../auth';
+import { detectOsIdentifier } from '../../helpers';
+import Tracking from '../../tracking';
+
+// wget --auth-no-challenge --user "$HOSTED_MENDER_EMAIL" --password "$HOSTED_MENDER_PASSWORD" https://downloads.customer.mender.io/content/hosted/mender-monitor/debian/1.2.0/mender-monitor_1.2.0-1%2Bdebian%2Bbuster_all.deb
+
+const cookies = new Cookies();
+
+const osMap = {
+  MacOs: 'darwin',
+  Unix: 'linux',
+  Linux: 'linux'
+};
+
+const defaultArchitectures = ['armhf', 'arm64', 'amd64'];
+const defaultOSVersions = ['debian+buster', 'debian+bullseye', 'ubuntu+bionic', 'ubuntu+focal'];
+
+const getVersion = (versions, id) => versions[id] || 'master';
+
+const getToolLocation = ({ isPublic = true }) => (isPublic ? 'mender.io' : 'customer.mender.io/content/hosted');
+
+const defaultLocationFormatter = ({ os, tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { id, osList = [], title } = tool;
+  if (osList.length) {
+    return osList.reduce((accu, supportedOs) => {
+      const title = Object.entries(osMap).find(entry => entry[1] === supportedOs)[0];
+      accu.push({
+        location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/${supportedOs}/${id}`,
+        title,
+        isUserOs: osMap[os] === supportedOs
+      });
+      return accu;
+    }, []);
+  }
+  return [{ location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/linux/${id}`, title }];
+};
+
+const multiArchLocationFormatter = ({ tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { archList, id, osList, packageId } = tool;
+  return osList.reduce((accu, os) => {
+    const osArchitectureLocations = archList.map(arch => ({
+      location: `https://downloads.${location}/repos/debian/pool/main/m/${id}/${encodeURIComponent(
```

Comment:
The `/m/` in the path is just the first letter of the package name (aka `${id}`). This works today because all our packages start by `m` (_m_ ender-connect, _m_ ender-client, ...). To be better future ready we should deduce it from `id`.

Reply:
mandatory: https://martinfowler.com/bliki/Yagni.html - it will now be deducted 👌